share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-rhel/spack.yaml

```diff
@@ -21,47 +21,156 @@ spack:
       target: [zen4]
       variants: +mpi
 
+    tbb:
+      require: "intel-tbb"
     binutils:
       variants: +ld +gold +headers +libiberty ~nls
+    boost:
+      variants: +python +filesystem +iostreams +system
+    cuda:
+      version: [11.7.0]
+    elfutils:
+      variants: +bzip2 ~nls +xz
+      require: "%gcc"
     hdf5:
       variants: +fortran +hl +shared
+    libfabric:
+      variants: fabrics=sockets,tcp,udp,rxm
     libunwind:
       variants: +pic +xz
+    mpich:
+      variants: ~wrapperrpath
     ncurses:
-      require: '@6.3 +termlib'
-    openblas:
-      require: '@0.3.20'
-      variants: threads=openmp
+      variants: +termlib
+    paraview:
+      # Don't build GUI support or GLX rendering for HPC/container deployments
+      require: "@5.11 ~qt+osmesa"
+    python:
+      version: [3.8.13]
+    trilinos:
+      require:
+      - one_of: [+amesos +amesos2 +anasazi +aztec +boost +epetra +epetraext +ifpack
+            +intrepid +intrepid2 +isorropia +kokkos +minitensor +nox +piro +phalanx
+            +rol +rythmos +sacado +stk +shards +stratimikos +tempus +tpetra
+            +trilinoscouplings +zoltan]
+      - one_of: [gotype=long_long, gotype=all]
+      - one_of: [~ml ~muelu ~zoltan2 ~teko, +ml +muelu +zoltan2 +teko]
+      - one_of: [+superlu-dist, ~superlu-dist]
+      - one_of: [+shylu, ~shylu]
```

Comment:
I was looking at other E4S stacks a couple of days ago, and noticed the same. Aren't these two lines redundant (requiring a Boolean variant to be either true or false)? What was their purpose when they were added?

Reply:
I don't recall. There was a concretization issue and this actually helped resolve it at one point. All of the `one_of` were added together to resolve the issue in the past. Perhaps it is not needed anymore? If you say these two lines have no effect, let us remove them and see what happens?